aidial_adapter_openai/utils/image_tokenizer.py

```diff
@@ -58,18 +54,20 @@ def _compute_high_detail_tokens(self, width: int, height: int) -> int:
     low_detail_tokens=2833, tokens_per_tile=5667
 )
 
-_TOKENIZERS: List[Tuple[ImageTokenizer, List[str]]] = [
-    (GPT4O_IMAGE_TOKENIZER, GPT4O_DEPLOYMENTS),
-    (GPT4O_MINI_IMAGE_TOKENIZER, GPT4O_MINI_DEPLOYMENTS),
-    (GPT4_VISION_IMAGE_TOKENIZER, GPT4_VISION_DEPLOYMENTS),
-]
-
 
-def get_image_tokenizer(deployment_id: str) -> ImageTokenizer | None:
-    for tokenizer, ids in _TOKENIZERS:
-        if deployment_id in ids:
-            return tokenizer
-    return None
+def get_image_tokenizer(
+    deployment_id: str, app_config: ApplicationConfig
+) -> ImageTokenizer:
+    if deployment_id in app_config.GPT4O_DEPLOYMENTS:
+        return GPT4O_IMAGE_TOKENIZER
+    elif deployment_id in app_config.GPT4O_MINI_DEPLOYMENTS:
+        return GPT4O_MINI_IMAGE_TOKENIZER
+    elif deployment_id in app_config.GPT4_VISION_DEPLOYMENTS:
+        return GPT4_VISION_IMAGE_TOKENIZER
+    else:
+        raise RuntimeError(
+            f"No image tokenizer found for deployment {deployment_id}"
+        )
```

Comment:
This new `RuntimeError` branch is unreachable, but we don't have a way to prove it statically.

The solution could be an introduction of a deployment classifier:

```
class ChatCompletionDeploymentType(str, Enum):
    DALLE3 = "DALLE3"
    MISTRAL = "MISTRAL"
    DATABRICKS = "DATABRICKS"
    GPT4_VISION = "GPT4_VISION"
    GPT4O = "GPT4O"
    GPT4O_MINI = "GPT4O_MINI"
    GPT_TEXT_ONLY = "GPT_TEXT_ONLY"
```

We could exhaustively match this enum and select an appropriate adapter.

Reply:
Done, refactored `call_chat_completion` with `match/case`